src/interface.jl

```diff
@@ -46,7 +46,7 @@ julia> StructArrays.createinstance(Complex{Float64}, (re=1.0, im=2.0)...)
 ```
 """
 function createinstance(::Type{T}, args...) where {T}
```

Comment:
```suggestion
function createinstance(::Type{T}, args...)::T where {T}
```
@piever @aplavin would doing this settle things?

This way `createinstance` will attempt to convert whatever it gets to a `T` and if that fails it throws and error. Here is how those asserts work:
```julia
julia> function foo(::Type{T}, x)::T where {T}
           x
       end
foo (generic function with 2 methods)

julia> foo(Int, 1)
1

julia> foo(Float64, 1)
1.0

julia> foo(Int, 1.0)
1

julia> foo(Int, 1.1)
ERROR: InexactError: Int64(1.1)
Stacktrace:
 [1] Int64
   @ ./float.jl:900 [inlined]
 [2] convert
   @ ./number.jl:7 [inlined]
 [3] foo(#unused#::Type{Int64}, x::Float64)
   @ Main ./REPL[32]:2
 [4] top-level scope
   @ REPL[36]:1
```

Reply:
🤷  if it's upsetting to you, we can instead 
```julia
convert(T, isconcretetype(T) ? bypass_constructor(T, args) : constructorof(T)(args...)) :: T
```
which is equivalent, but uglier. I just think it'd be good to settle on something since you seemed to really want this PR to merge, and this seemed like a very minimal way to fix @piever's concerns while still taking advantage of `constructorof`.

Reply:
Or maybe I'm just reading too much into your usage "...", sometimes that means different things to different people